src/components/StartupViewProvider.ts

```diff
@@ -29,13 +29,18 @@ export class StartupViewProvider implements vscode.WebviewViewProvider {
     webviewView.webview.html = this._getHtmlForWebview(webviewView.webview, this.viewType)
 
     webviewView.webview.onDidReceiveMessage(async (data: { btnType: BUTTON_TYPES }) => {
+      const { shouldShowCliStartUpView, shouldShowWorkspaceView } = await this.checkView()
```

Comment:
rather than doing both of these for any button click, we could just run the check relevant to the clicked button